packages/ap-pdf/alliance_platform/pdf/render.py

```diff
@@ -0,0 +1,147 @@
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+
+from alliance_platform.frontend.bundler import get_bundler
+from alliance_platform.frontend.bundler.vite import ViteBundler
+from alliance_platform.pdf.render_process import process_request
+from alliance_platform.pdf.request_handlers import CustomRequestHandler
+from alliance_platform.pdf.request_handlers import DjangoRequestHandler
+from alliance_platform.pdf.request_handlers import MediaHttpRequestHandler
+from alliance_platform.pdf.request_handlers import PassThroughRequestHandler
+from alliance_platform.pdf.request_handlers import RequestHandler
+from alliance_platform.pdf.request_handlers import StaticHttpRequestHandler
+from alliance_platform.pdf.request_handlers import WhitelistDomainRequestHandler
+from alliance_platform.pdf.settings import ap_pdf_settings
+from django.conf import settings
+
+RENDERER_PROCESS = os.path.abspath(os.path.join(os.path.dirname(__file__), "render_process.py"))
+
+
+class PDFRendererException(Exception):
+    """Errors in PDF rendering will be communicated via this exception type."""
+
+
+def get_default_handlers() -> list[RequestHandler]:
+    """Returns the default request handlers to use
+
+    Includes :class:`~alliance_platform.pdf.request_handlers.StaticHttpRequestHandler`, :class:`~alliance_platform.pdf.request_handlers.MediaHttpRequestHandler`,
+    :class:`~alliance_platform.pdf.request_handlers.DjangoRequestHandler` and :class:`~alliance_platform.pdf.request_handlers.WhitelistDomainRequestHandler`.
+
+    :class:`~alliance_platform.pdf.request_handlers.WhitelistDomainRequestHandler` will use domains in the :py:attr:`~alliance_platform.pdf.settings.AlliancePlatformPDFSettingsType.WHITELIST_DOMAINS`
+    setting and, when ``DEBUG=True``, also include the Vite dev server URL as extracted from the stats file.
+    """
+
+    whitelist_domains = [*ap_pdf_settings.WHITELIST_DOMAINS]
+    bundler = get_bundler()
+    if isinstance(bundler, ViteBundler) and bundler.is_development():
+        whitelist_domains.append(bundler.dev_server_url)
```

Comment:
this adds a dependency on the frontend package, we should probably make this something you have to add manually. i'm not entirely sure if we can just do it in settings, or if it needs to be dynamic after apps are ready? either way I think it would be good to have no dependency on alliance_platform_frontend

Reply:
or maybe just make it optional and conditional on alliance_platform_frontend being installed?